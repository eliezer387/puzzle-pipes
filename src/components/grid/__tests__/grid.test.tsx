import React from "react";
import { render, fireEvent } from '@testing-library/react';
import Grid from "..";


describe('Grid', () => {

    it('render grid', () => {
        const elements =[["1","2","3"],["4","5","6"],["7","8","9"]];
        const rotate = jest.fn();
        const { getByText } = render(<Grid elements={elements}  rotate={rotate} />);
        const linkElement = getByText(/4/i);
        expect(linkElement).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const elements =[["1","2","3"],["4","5","6"],["7","8","9"]];
        const rotate = jest.fn();
        const { asFragment } = render(<Grid elements={elements}  rotate={rotate} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('click on grid', () => {
        const elements =[["1","2","3"],["4","5","6"],["7","8","9"]];
        const rotate = jest.fn();
        const { getByText } = render(<Grid elements={elements}  rotate={rotate} />);
        const linkElement = getByText(/4/i);
        fireEvent.click(linkElement);
        expect(rotate).toHaveBeenCalled();
    });
})