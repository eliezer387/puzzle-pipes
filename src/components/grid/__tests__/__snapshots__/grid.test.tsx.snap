// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL ginSla"
  >
    <div
      class="sc-gsDKAQ bGbxpt"
    >
      <div
        class="sc-dkPtRN kgUzps"
      >
        1
      </div>
      <div
        class="sc-dkPtRN kgUzps"
      >
        2
      </div>
      <div
        class="sc-dkPtRN kgUzps"
      >
        3
      </div>
    </div>
    <div
      class="sc-gsDKAQ bGbxpt"
    >
      <div
        class="sc-dkPtRN kgUzps"
      >
        4
      </div>
      <div
        class="sc-dkPtRN kgUzps"
      >
        5
      </div>
      <div
        class="sc-dkPtRN kgUzps"
      >
        6
      </div>
    </div>
    <div
      class="sc-gsDKAQ bGbxpt"
    >
      <div
        class="sc-dkPtRN kgUzps"
      >
        7
      </div>
      <div
        class="sc-dkPtRN kgUzps"
      >
        8
      </div>
      <div
        class="sc-dkPtRN kgUzps"
      >
        9
      </div>
    </div>
  </div>
</DocumentFragment>
`;
